model_analyzer/config/input/config_command_profile.py

```diff
@@ -178,6 +178,14 @@ def _fill_config(self):
                 default_value=DEFAULT_DURATION_SECONDS,
                 description=
                 'Specifies how long (seconds) to gather server-only metrics'))
+        self._add_config(
+            ConfigField(
+                'metrics',
```

Comment:
Would this end up cleaner if we made this a boolean flag that only determines whether we collect cpu metrics or not? Much of the logic here would stay the same, except the challenge would only be removing CPU metrics from the reports and tables etc unless this value is true.

Reply:
If we are going back to the on/off mode, should we also bring back the `auto` mode?
`True`: Collect CPU metrics
`False`: Do not collect CPU metrics
`auto`: Collect CPU metrics iff the model will run on CPU. (default to)
Then, typical users would not need to deal with this flag, except they want or do not want CPU metrics.

Reply:
@Tabrizian @kthui what are your thoughts on this. If we don't want to go back to a boolean flag `--collect-cpu-metrics`, then the rest of this PR looks good to me.

Reply:
A summary of the changes so far:
  - `--metrics` flag of the `profile` stage specifies what metrics are to be collected.
  - A minimum set of metrics is enforced similar to `inference_output_fields`, ... of the `analyze` stage.
  - The 2 CPU metrics are currently optional.
  - `analyze` stage will find the union of metrics collected during `profile`, instead of using a fixed list of metrics.
  - By default, CPU metrics are not included in both `profile` and `analyze` stages.
  - A unit test is added to check the CPU monitor will not run if non of the CPU metrics are requested during `profile` stage.
  - L0_metrics test is added to check requesting different metrics works.

I think this change is more extensible than a boolean flag, in the future.

Reply:
Looks good to me as well. If the default run of MA is warning free, then the changes look good to me.

Reply:
CPU metrics are not included by default during `analyze` stage (and they are not collected by default), so it is error free.